contrib/doc/pinmap/generate.py

```diff
@@ -0,0 +1,1239 @@
+import io
```

Comment:
This is impressive.

Maybe throw a shebang at the top and `chmod` the file `0755` for us *nix users.

```suggestion
#!/usr/bin/env python3

import io
```